epp-client/src/epp/request.rs

```diff
@@ -5,15 +5,62 @@ pub mod domain;
 pub mod host;
 pub mod message;
 
-use serde::{ser::SerializeStruct, ser::Serializer, Deserialize, Serialize};
+use epp_client_macros::*;
+
+use serde::{de::DeserializeOwned, ser::SerializeStruct, ser::Serializer, Deserialize, Serialize};
 use std::error::Error;
+use std::fmt::Debug;
 use std::time::SystemTime;
 
 use crate::epp::object::{
     ElementName, EmptyTag, EppObject, Extension, Options, ServiceExtension, Services, StringValue,
 };
-use crate::epp::xml::{EPP_CONTACT_XMLNS, EPP_DOMAIN_XMLNS, EPP_HOST_XMLNS, EPP_LANG, EPP_VERSION};
-use epp_client_macros::*;
+use crate::epp::response::{CommandResponseStatus, CommandResponseWithExtension};
+use crate::epp::xml::{
+    EppXml, EPP_CONTACT_XMLNS, EPP_DOMAIN_XMLNS, EPP_HOST_XMLNS, EPP_LANG, EPP_VERSION,
+};
+
+type DeserializeResult<T> = Result<
```

Comment:
The compiler was complaining about type complexity in the function signature. Not sure if there is a more idiomatic way to do this.